File: Round-1/Array/Decompress Run-Length Encoded List/sol.js

var decompressRLElist = function (nums) {
    const decodedList = [];
    for (let i = 0; i < nums.length; i += 2) {
        for (let j = 0; j < nums[i]; j++) {
            decodedList.push(nums[i + 1])
        }
    }
    return decodedList;
};

decompressRLElist([1, 2, 3, 4])

File: Round-1/Array/others/Decompress Run-Length Encoded List/sol.cpp
// Sol - 1 >
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> decompressRLElist(vector<int> &nums)
    {
        vector<int> decodedList;
        for (int32_t i = 0; i < nums.size(); i += 2)
        {
            for (int32_t j = 0; j < nums[i]; ++j)
            {
                decodedList.push_back(nums[i + 1]);
            }
        }
        return decodedList;
    }
};
int main()
{
    vector<int> Sol_Array{1, 2, 3, 4};
    Solution Sol;
    Sol.decompressRLElist(Sol_Array);
    return 0;
}